src/foam/core/AbstractPropertyInfo.java

```diff
@@ -157,7 +157,7 @@ public void authorize(X x) {
       AuthService auth = (AuthService) x.get("auth");
       String simpleName = this.getClassInfo().getObjClass().getSimpleName();
 
-      if ( ! auth.check(x, String.format("%s.ro.%s", simpleName.toLowerCase(), this.getName().toLowerCase())) ) {
+      if ( ! auth.check(x, String.format("%s.ro.%s", simpleName.toLowerCase(), this.getName().toLowerCase())) && ! auth.check(x, String.format("%s.rw.%s", simpleName.toLowerCase(), this.getName().toLowerCase()))) {
```

Comment:
I think just concatenating the strings with + is much faster than using String.format().

Reply:
Don't call toLowerCase() twice on the same values. Store in variables.